examples/multi_gpu/mag240m_graphsage.py

```diff
@@ -0,0 +1,422 @@
+    if rank == 0:
+        print("Setting up GNN...")
+    acc = Accuracy(task="multiclass", num_classes=data.num_classes)
+    in_channels = data["paper"].x.size(-1)
+    model = GraphSAGE(
+        in_channels=in_channels,
+        hidden_channels=hidden_channels,
+        num_layers=len(sizes),
+        out_channels=data.num_classes,
+        dropout=dropout,
+        metadata=data.metadata(),
+    )
+
+    if rank == 0:
+        print(f"# GNN Params: \
+            {sum([p.numel() for p in model.parameters()])/10**6:.1f}M")
+        print('Setting up NeighborLoaders...')
+    train_idx = data["paper"].train_mask.nonzero(as_tuple=False).view(-1)
+    eval_idx = data["paper"].val_mask.nonzero(as_tuple=False).view(-1)
+    if n_devices > 1:
+        # Split indices into `n_devices` many chunks:
+        train_idx = train_idx.split(train_idx.size(0) // n_devices)[rank]
+        eval_idx = eval_idx.split(eval_idx.size(0) // n_devices)[rank]
+
+    # delete unused tensors to not sample
+    del data["paper"].train_mask
+    del data["paper"].val_mask
+    del data["paper"].test_mask
+    del data["paper"].year
+
+    kwargs = dict(batch_size=batch_size, num_workers=16,
+                  persistent_workers=True, num_neighbors=sizes, drop_last=True)
+    train_loader = NeighborLoader(
+        data,
+        input_nodes=("paper", train_idx),
+        shuffle=True,
+        **kwargs,
+    )
+
+    eval_loader = NeighborLoader(
+        data,
+        input_nodes=("paper", eval_idx),
+        shuffle=True,
+        **kwargs,
+    )
+
+    # original OGB example also tests on eval_idx
+    # it saves test_idx for the final hidden test
+    # for the OGB competition.
+    test_loader = NeighborLoader(
+        data,
+        input_nodes=("paper", eval_idx),
+        **kwargs,
+    )
+
+    if rank == 0:
+        print("Final setup...")
+    if n_devices > 0:
+        model = model.to(rank)
+        acc = acc.to(rank)
+    if rank == 0:
+        print("about to make optimizer")
+    if n_devices > 1:
+        model = DistributedDataParallel(model, device_ids=[rank])
+    optimizer = torch.optim.Adam(model.parameters(), lr=lr)
+    if rank == 0:
+        print("Beginning loop...")
+    for epoch in range(1, num_epochs + 1):
+        model.train()
+        time_sum, acc_sum = 0, 0
+        for i, batch in enumerate(train_loader):
+            if num_steps_per_epoch >= 0 and i >= num_steps_per_epoch:
+                break
+            if i >= num_warmup_iters_for_timing:
+                torch.cuda.synchronize()
+                since = time.time()
+            optimizer.zero_grad()
+
+            if n_devices > 0:
+                batch = batch.to(rank, "x", "y", "edge_index")
+                # Features loaded in as fp16, train in 32bits
+                batch['paper'].x = batch['paper'].x.to(torch.float32)
+            loss, train_acc = training_step(batch, acc, model)
+            acc_sum += train_acc
```

Comment:
`torchmetrics` holds states in the `Metric` class, so afaik, you don't need to manually accumulate the accuracy yourself. Might be worth looking at their docs here: https://lightning.ai/docs/torchmetrics/stable/pages/quickstart.html#module-metrics